### DISCUSSION TITLE
File-based access to documents (e.g. via WebDAV)

### DISCUSSION THREAD

**Discussion by martiniusDE** [jonaswinkler/paperless-ng#839] [Feature Requests]
Hi, thanks for the effort building paperless-ng!

It would be really helpful if it would be possible to access the documents with a directory/file structure via third party programs, like Windows Explorer or Mac OS Finder. So if I connect via WebDAV for example, it would be great to:

- access the files in a folder structure
- structure should be _dynamic_ based on types, tags, corrospondents; some ideas:
  - ${RootFolder}/${documentType}/${tags}/FileName.pdf
  - ${RootFolder}/${corrospondent}/${dates[2020/09/]}/FileName.pdf
  - ...
  - Filename should be created automatically (e.g. "2020-09-03 $corrospondent $type") and/or edited via web ui ("title" or so)
- read-only would be enough, but could be editable too (if so, the documents should be processed if added/updated recognizing the folder name for tagging)

Thanks in advance!

Best, Martin

**Comment by benjaminfrank** [comment]
While not exactly what you want you can leverage the `PAPERLESS_FILENAME_FORMAT` setting for one variant of naming schema and then use samba to make the files available to others. That is how I do it at the moment. If you want WebDAV you could use owncloud/nextcloud to make files available.
I am not sure if modifying the files is a good idea though, renaming most likely interferes with the paperles-ng DB.

**Reply by jonaswinkler** [reply]
> I am not sure if modifying the files is a good idea though, renaming most likely interferes with the paperles-ng DB.

Exactly. If you manually move files, paperless wont be able to find them and report them as missing.

**Reply by martiniusDE** [reply]
Thanks for your suggestions. I think read-only would be totally fine. What I mean is this: 

It would be really helpful if the files are sorted (dynamically, through paperless-ng webdav, not samba) in directories based on content/category/tags or so and the files have some information in the file name (e. g. date, contact, tag - somtething related to the actual contet). Maybe it has to be configurable in the ui. That way, someone could easily find documents based in content. I know that this means some work, but would make it a kind of a DMS.

**Comment by stefan2904** [comment]
I think you can just use the paperless media folder and make that available (preferably read only) via standard Linux means? 

Or the other way around, set the media folder to a smb share, webdav of your Nextcloud, or whatever?
